eZ/Publish/Core/Persistence/Legacy/Content/Gateway/DoctrineDatabase.php

```diff
@@ -1546,488 +1180,353 @@ protected function removeRelationFromRelationListField($contentId, array $row)
+            ->setParameter('content_id', $contentId, ParameterType::INTEGER)
+            ->setParameter('version_no', $version, ParameterType::INTEGER)
+            ->setParameter('language_id', $language->id, ParameterType::INTEGER)
+            ->setParameter('language_code', $language->languageCode, ParameterType::STRING)
+        ;
+
+        if (!$this->contentNameExists($contentId, $version, $language->languageCode)) {
+            $query
+                ->insert(self::CONTENT_NAME_TABLE)
+                ->values(
+                    [
+                        'contentobject_id' => ':content_id',
+                        'content_version' => ':version_no',
+                        'content_translation' => ':language_code',
+                        'name' => ':name',
+                        'language_id' => $this->getSetNameLanguageMaskSubQuery(),
+                        'real_translation' => ':language_code',
+                    ]
+                );
+        } else {
+            $query
+                ->update(self::CONTENT_NAME_TABLE)
+                ->set('name', ':name')
+                ->set('language_id', $this->getSetNameLanguageMaskSubQuery())
+                ->set('real_translation', ':language_code')
+                ->where('contentobject_id = :content_id')
+                ->andWhere('content_version = :version_no')
+                ->andWhere('content_translation = :language_code');
+        }
 
-        $query->prepare()->execute();
+        $query->execute();
     }
 
     /**
-     * Loads relations from $contentId to published content, optionally only from $contentVersionNo.
+     * Return a language sub select query for setName.
      *
-     * $relationType can also be filtered.
-     *
-     * @param int $contentId
-     * @param int $contentVersionNo
-     * @param int $relationType
+     * The query generates the proper language mask at the runtime of the INSERT/UPDATE query
+     * generated by setName.
      *
-     * @return string[][] array of relation data
+     * @see setName
      */
-    public function loadRelations($contentId, $contentVersionNo = null, $relationType = null)
+    private function getSetNameLanguageMaskSubQuery(): string
     {
-        $query = $this->queryBuilder->createRelationFindQuery();
-        $query->innerJoin(
-            $query->alias(
-                $this->dbHandler->quoteTable('ezcontentobject'),
-                'ezcontentobject_to'
-            ),
-            $query->expr->lAnd(
-                $query->expr->eq(
-                    $this->dbHandler->quoteColumn('to_contentobject_id', 'ezcontentobject_link'),
-                    $this->dbHandler->quoteColumn('id', 'ezcontentobject_to')
-                ),
-                $query->expr->eq(
-                    $this->dbHandler->quoteColumn('status', 'ezcontentobject_to'),
-                    $query->bindValue(1, null, \PDO::PARAM_INT)
+        return <<<SQL
```

Comment:
> since php 7.3 heredocs dont have to be formatted to left side, I think it would look better ;)

Squashed into 8d42846 (linking to lines of collapsed files seems to be broken, sorry) and aligned the remaining one  via 0390e6b.